kedro-datasets/pyproject.toml

```diff
@@ -198,25 +199,28 @@ test = [
   "jupyter~=1.0",
   "lxml~=4.6",
   "matplotlib>=3.0.3, <3.4; python_version < '3.10'",  # 3.4.0 breaks holoviews
-  "matplotlib>=3.5, <3.6; python_version == '3.10'",
+  "matplotlib>=3.5, <3.6; python_version >= '3.10'",
   "memory_profiler>=0.50.0, <1.0",
   "moto==1.3.7; python_version < '3.10'",
-  "moto==3.0.4; python_version == '3.10'",
+  "moto==4.1.12; python_version >= '3.10'",
   "networkx~=2.4",
   "opencv-python~=4.5.5.64",
   "openpyxl>=3.0.3, <4.0",
-  "pandas-gbq>=0.12.0, <0.18.0",
+  "pandas-gbq>=0.12.0, <0.18.0; python_version < '3.11'",
+  "pandas-gbq>=0.18.0; python_version >= '3.11'",
   "pandas>=1.3, <2",  # 1.3 for read_xml/to_xml, <2 for compatibility with Spark < 3.4
   "Pillow~=9.0",
   "plotly>=4.8.0, <6.0",
   "polars~=0.15.13",
   "pre-commit>=2.9.2, <3.0",  # The hook `mypy` requires pre-commit version 2.9.2.
   "psutil==5.8.0",
-  "pyarrow~=8.0",
+  "pyarrow>=1.0; python_version < '3.11'",
+  "pyarrow>=7.0; python_version >= '3.11'",  # Adding to avoid numpy build errors
   "pylint>=2.5.2, <3.0",
   "pyodbc~=4.0.35",
   "pyproj~=3.0",
-  "pyspark>=2.2, <4.0",
+  "pyspark>=2.2, <3.4; python_version < '3.11'",
+  "pyspark>=3.4; python_version >= '3.11'",
```

Comment:
Ah so because our tests force it to installed >=3.4, we should update the `spark-base` definition.